src/lib/y2storage/clients/partitions_proposal.rb

```diff
@@ -137,12 +139,26 @@ def failed_proposal
 
       def successful_proposal
         {
-          "preformatted_proposal" => actions_presenter.to_html,
+          "preformatted_proposal" => preformatted_proposal,
           "links"                 => actions_presenter.events,
           "language_changed"      => false
         }
       end
 
+      def preformatted_proposal
+        return nil if @failed
```

Comment:
It is safe/expected to return `nil` here?

Reply:
yes, this is just for consistence; it is not called from failed_proposal, but this is the value failed_proposal returns for this field.

Reply:
See
https://github.com/yast/yast-storage-ng/blob/master/src/lib/y2storage/clients/partitions_proposal.rb#L130